examples/demo/client/Dockerfile

```diff
@@ -1,6 +1,6 @@
 # Copyright The OpenTelemetry Authors
 # SPDX-License-Identifier: Apache-2.0
-FROM golang:1.20 as build
+FROM golang:1.20.7 as build
```

Comment:
@dmitryax should I also revert this and other such changes?

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/25162#discussion_r1292834345

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/25162#discussion_r1292834906

Reply:
Yes please. We should not define patch version of the docker image